README.md

```diff
@@ -38,41 +38,56 @@ var parentWalletPublicKey = parentWalletSecret.slice(64, 128)
 var childWalletSecret = lib.derivePublic(parentWalletPublicKey, 1, 1)
 ```
 
-# available functions
+# Available Functions
 
+## Signing/Verification
 * `Buffer sign(Buffer msg, Buffer walletSecret)`
 * `Bool verify(Buffer msg, Buffer publicKey, Buffer sig)`
+
+## Key derivation
 * `async Buffer mnemonicToRootKeypair(String mnemonic, int derivationScheme)`
 * `Buffer derivePrivate(Buffer parentKey, int index, int derivationScheme)`
 * `Buffer derivePublic(Buffer parentExtPubKey, int index, int derivationScheme)`
 * `Buffer toPublic(Buffer privateKey)`
-* `Buffer decodePaperWalletMnemonic(string paperWalletMnemonic)`
-* `Buffer xpubToHdPassphrase(Buffer xpub)`
-* `Buffer packBootstrapAddress(Array[int] derivationPath, Buffer xpub, Buffer hdPassphrase, int derivationScheme)`
-* `Buffer packBaseAddress(Buffer pubKey, Buffer stakePubKey, int addressType, int networkId, Bool isStakeHash = false)`
-* `Buffer packPointerAddress(Buffer pubKey, Object pointer, int addressType, int networkId)`
-* `Buffer packEnterpriseAddress(Buffer pubKey, int addressType, int networkId)`
-* `Buffer packRewardsAccountAddress(Buffer stakePubkey, int addressType, int networkId, Bool isStakeHash = false)`
-* `Object getAddressInfo(Buffer address)`
+
+## Address encoding/decoding/validation
+* `Buffer packBootstrapAddress(Array[int] derivationPath, Buffer xpub, Buffer hdPassphrase, int derivationScheme, int protocolMagic)`
+* `Buffer packBaseAddress(Buffer spendingKeyHash, Buffer stakingPubKey, int networkId)`
+* `Buffer packPointerAddress(Buffer pubKeyHash, Object pointer, int networkId)`
+* `Buffer packEnterpriseAddress(Buffer spendingKeyHash, int networkId)`
+* `Buffer packRewardAddress(Buffer stakingKeyHash, int networkId)`
+* `Object getAddressType(Buffer address)`
+* `Object getShelleyAddressInfo(Buffer address)`
 * `Object AddressTypes`
-* `string unpackAddress(string address, Buffer hdPassphrase)`
-* `Bool isValidAddress(string address)`
+* `Buffer addressToBuffer(string address) // address can be either bech32 or base58 encoded`
+* `Map getBootstrapAddressAttributes(Buffer address)`
+* `Array<int>? getBootstrapAddressDerivationPath(Buffer address, Buffer hdPassphrase)`
+* `int getBootstrapAddressProtocolMagic(Buffer address)`
+* `Bool isValidBootstrapAddress(string address)`
```

Comment:
it probably makes sense to expose a generic `isValidAddress` call which would use the correct validation logic depending on the address type (bootstrap/shelley)